dev/Repeater/APITests/RepeaterTests.cs

```diff
@@ -234,6 +234,78 @@ public void NestedRepeaterWithDataTemplateScenario()
             NestedRepeaterWithDataTemplateScenario(disableAnimation: false);
         }
 
+
+        [TestMethod]
+        public void FocusedItemGetsRecycledUponCollectionReset()
+        {
+            List<object> ResettingListItems = new List<object> { "item0", "item1", "item2", "item3", "item4", "item5", "item6", "item7", "item8", "item9" };
+            ItemsRepeater repeater;
+            RunOnUIThread.Execute(() => {
+
+                var template = (DataTemplate)XamlReader.Load(
+                        @"<DataTemplate xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'> 
+                            <Button AutomationProperties.Name='{Binding}' Content='{Binding}'/>
+                        </DataTemplate>");
+
+                repeater = new ItemsRepeater() {
+                    ItemsSource = ResettingListItems,
+                    ItemTemplate = template
+                };
+
+                Content = new ItemsRepeaterScrollHost() {
+                    Width = 400,
+                    Height = 800,
+                    ScrollViewer = new ScrollViewer {
+                        Content = repeater
+                    }
+                };
+
+                Content.UpdateLayout();
+                
+                Random r = new Random();
+                int index = r.Next(1, 10);
+                string removedText = "item" + index;
+
+                Log.Comment("Focusing a button");
+                Button toFocus = (Button)repeater.TryGetElement(index);
+                Verify.AreEqual(removedText,toFocus.Content as string);
+                toFocus.Focus(FocusState.Programmatic);
+                IdleSynchronizer.Wait();
```

Comment:
Close the UIThread scope and call IdleSynchronizer outside the UIThread.

Reply:
Fixed that. However the unit test is still not failing without the fix.

Reply:
I see. Taking a look.

Reply:
I pushed some changes to clean it up and get it to work. Turns out the real issue is that a previous change broke the Verify methods - so they the tests are just logging failure but not failing. Following up on that. You can debug into the verify methods to see that.

Reply:
We have a fix for the test not failing issue. So this should work now.

Reply:
OH I see. Thank you for taking your time and fixing the issue!